src/app/modules/service-providers/services/routes/routes.service.ts

```diff
@@ -32,9 +65,10 @@ export class RoutesService {
       try {
         const routeBase = result.data.resourceSets[0].resources[0];
         routeDetails.routePoints = routeBase.routePath.line.coordinates;
-        routeDetails.routeLength = routeBase.travelDistance;
+        routeDetails.routeLength = length(lineString(routeDetails.routePoints)) * 1000;
         routeDetails.routeDuration = routeBase.travelDuration;
-        routeDetails.routeLegs = indexLegs(routeBase.routeLegs);
+        routeDetails.routeLegs = mapLegs(routeBase.routeLegs[0].itineraryItems);
+        routeDetails.routeLegs = indexLegs(routeDetails.routeLegs, routeDetails.routePoints);
```

Comment:
indexLegs just adds/update the value, it's shouldn't override the routeLegs. i.e. routeLegs should be created one